pkg/apis/core/validation/validation.go

```diff
@@ -5638,6 +5638,11 @@ func validateKubeFinalizerName(stringValue string, fldPath *field.Path) field.Er
 // newNamespace is updated with fields that cannot be changed
 func ValidateNamespaceUpdate(newNamespace *core.Namespace, oldNamespace *core.Namespace) field.ErrorList {
 	allErrs := ValidateObjectMetaUpdate(&newNamespace.ObjectMeta, &oldNamespace.ObjectMeta, field.NewPath("metadata"))
+	if utilfeature.DefaultFeatureGate.Enabled(features.NamespaceDefaultLabelName) {
+		if len(newNamespace.Name) > 0 && newNamespace.Labels[v1.LabelMetadataName] != newNamespace.Name {
```

Comment:
generateName should be resolved by the time we get here, so no need to check len()

Reply:
also add to ValidateNamespaceStatusUpdate and ValidateNamespaceFinalizeUpdate

Reply:
fixd